messaging/src/main/java/org/axonframework/eventhandling/annotation/AnnotatedEventHandlingComponent.java

```diff
@@ -0,0 +1,137 @@
+     *                              {@link org.axonframework.eventhandling.annotation.EventHandler} annotated methods.
+     */
+    public AnnotatedEventHandlingComponent(@Nonnull T annotatedEventHandler) {
+        this(annotatedEventHandler, ClasspathParameterResolverFactory.forClass(annotatedEventHandler.getClass()));
+    }
+
+    /**
+     * Wraps the given {@code annotatedEventHandler}, allowing it to be subscribed to a
+     * {@link org.axonframework.eventhandling.EventSink} as a {@link EventHandlingComponent}.
+     *
+     * @param annotatedEventHandler    The object containing the
+     *                                 {@link org.axonframework.eventhandling.annotation.EventHandler} annotated
+     *                                 methods.
+     * @param parameterResolverFactory The strategy for resolving handler method parameter values.
+     */
+    public AnnotatedEventHandlingComponent(@Nonnull T annotatedEventHandler,
+                                           @Nonnull ParameterResolverFactory parameterResolverFactory) {
+        this(annotatedEventHandler,
+             parameterResolverFactory,
+             ClasspathHandlerDefinition.forClass(annotatedEventHandler.getClass()));
+    }
+
+    /**
+     * Wraps the given {@code annotatedEventHandler}, allowing it to be subscribed to a
+     * {@link org.axonframework.eventhandling.EventSink} as a {@link EventHandlingComponent}.
+     *
+     * @param annotatedEventHandler    The object containing the
+     *                                 {@link org.axonframework.eventhandling.annotation.EventHandler} annotated
+     *                                 methods.
+     * @param parameterResolverFactory The strategy for resolving handler method parameter values.
+     * @param handlerDefinition        The handler definition used to create concrete handlers.
+     */
+    @SuppressWarnings("unchecked")
+    public AnnotatedEventHandlingComponent(@Nonnull T annotatedEventHandler,
+                                           @Nonnull ParameterResolverFactory parameterResolverFactory,
+                                           @Nonnull HandlerDefinition handlerDefinition) {
+        this.target = requireNonNull(annotatedEventHandler, "The Annotated Event Handler may not be null");
+        this.model = AnnotatedHandlerInspector.inspectType((Class<T>) annotatedEventHandler.getClass(),
+                                                           parameterResolverFactory,
+                                                           handlerDefinition);
+    }
+
+    @Override
+    public EventHandlerRegistry subscribe(@Nonnull QualifiedName name, @Nonnull EventHandler eventHandler) {
+        throw new UnsupportedOperationException(
+                "This Command Handling Component does not support direct command handler registration."
+        );
+    }
+
+    @Nonnull
+    @Override
+    public MessageStream.Empty<Message<Void>> handle(@Nonnull EventMessage<?> event,
```

Comment:
After working on that, I'm not sure if it's OK that we enforce MessageStream.Empty as a result type.
It'd be more extensible if we allow Single here. Empty extends Single, so we can still return empty as we want. But many times when I was developing Event-Driven apps, I wanted to publish a command in reaction to an event. I can imagine if we allow Single, I'd be able to add an interceptor which publishes every command returned from an event handler (so maybe even just regular MessageStream - because I may want to publish many). Now I cannot do this (without using CommandGateway directly in every EventHandler). I'm not sure if it's beneficial that we limit it to just Empty.

Reply:
So, the result of event handling to be a command message? To be frank, that feels more like a side-effect of event handling then the result. Just as updating a query model is a side-effect. Or invoking a third-party service.

Could you per chance elaborate where you performed this flow and in what context it would be desirable? Not saying we shouldn't have that, but I am not certain at this stage whether a "command-message-retuning-event-message-handler" == `EventHandler`. Perhaps, it is something else.

Reply:
Hmmm... I think your're right. 
It's event handler used to integrate two contexts/aggregates. I don't expect any compensation actions, it's just a dump pipe. Like Ivan's Saga from: https://github.com/fraktalio/fmodel?tab=readme-ov-file#saga

I may always do an EventHandler like below on my own, so I agree we don't need to introduce the return type change.
```java
abstract class ReactEventHandler {
   private final EventSink eventSink;
   private final CommandGateway commandGateway;

  abstract Command react(Event event);

   @EventHandler
   void handle(event){
     var command = react(event);
     commandGateway.send(command);
   }

}
```